src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,380 @@
+     */
+    event Pick(address indexed who);
+    /**
+     * @notice `wad` amount of Dai was pushed to the recipient `to`.
+     * @param to The Dai recipient address
+     * @param wad The amount of Dai
+     */
+    event Push(address indexed to, uint256 wad);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. Currently the supported values are: "quitTo".
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The conduit outstanding gem balance was flushed out to `quitTo` address.
+     * @param quitTo The quitTo address.
+     * @param wad The amount flushed out.
+     */
+    event Quit(address indexed quitTo, uint256 wad);
+
+    /**
+     * @notice Defines PSM and quitTo addresses and gives `msg.sender` admin access.
+     * @param _psm PSM contract address.
+     */
+    constructor(address _psm) public {
+        GemAbstract _gem = GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem());
+        psm = PsmAbstract(_psm);
+        gem = _gem;
+        dai = DaiAbstract(PsmAbstract(_psm).dai());
+
+        uint256 gemDecimals = _gem.decimals();
+        uint256 daiDecimals = dai.decimals();
+        require(gemDecimals <= daiDecimals, "RwaOutputConduit3/invalid-gem-decimals");
+        to18ConvertionFactor = 10**(daiDecimals - gemDecimals);
+
+        // Give unlimited approve to PSM
+        dai.approve(_psm, type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaOutputConduit3/not-authorized");
+        _;
+    }
+
+    modifier isMate() {
+        require(may[msg.sender] == 1, "RwaOutputConduit3/not-mate");
+        _;
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /**
+     * @notice Grants `usr` operator access to this contract.
+     * @param usr The user address.
+     */
+    function hope(address usr) external auth {
```

Comment:
I think you can use `10 ** sub(dai.decimals(), _gem.decimals())` here and remove the 3 lines above

Reply:
I hadn't seen this comment, made the same one. But I think what we actually need to do is introducing back the `operator` and not messing with the roles.

Reply:
done